applications/FluidDynamicsApplication/custom_elements/alternative_d_vms_dem_coupled.cpp

```diff
@@ -170,11 +181,20 @@ void AlternativeDVMSDEMCoupled<TElementData>::Initialize(const ProcessInfo& rCur
     // Base class does things with constitutive law here.
     DVMS<TElementData>::Initialize(rCurrentProcessInfo);
 
+    if(Dim == 2){
+        if (NumNodes == 9 || NumNodes == 6 || NumNodes == 4)
+            mInterpolationOrder = 2;
+    }
+    else if(Dim == 3){
+        if (NumNodes == 10 || NumNodes == 27)
+            mInterpolationOrder = 2;
+    }
```

Comment:
As these are template parameters, in this case you can use `if constexpr` so you evaluate the if at compile time.

(I'd do this in a future PR considering that this is already taking a while...)